### DISCUSSION TITLE
How to hide crosshair?

### DISCUSSION THREAD

**Discussion by steffos** [Live-Charts/LiveCharts2#1254] [Q&A]
I have a .net maui solution where I use the cartesian chart control. 

The charts underlying data is changed by user actions and the chart rerenders just fine.

My chart supports crosshair - when user touches a point in the chart, a crosshair is shown.

Now I need to hide the crosshair every time I change the underlying data but I can't figure out how to do this. It seems that once the crosshair is shown, it stays there even on underlying data refresh and I can't seem to make it disappear. 

I did find an `InvalidateCrosshair` method on the `Axis` type but that one also takes a point that I don't know how to supply or if it is meant to function the way I want.

Anyone knows how to remove the crosshair on an existing chart control and then not show it until the next time the user taps the chart?

**Comment by beto-rodriguez** [comment]
I think we can improve this.... 

We should hide the crosshairs when the finger goes up (same as tooltips in `RC-1`)

I will move this to an issue so we can track the implementation of this feature, this need to be included in the next version of the library.

**Reply by beto-rodriguez** [reply]
Transfered to #1255

**Reply by steffos** [reply]
This sounds great but leads me to another question - in the .net maui solution I work with the crosshair only shows when user taps the chart (put down finger and lift it up). So user has to tap at every point they want the crosshair to be shown. I would love to have it shown only on swiping but I have not found a way to configure this. Is this configurable in the current version?

**Reply by beto-rodriguez** [reply]
No that is not configurable now.

> I would love to have it shown only on swiping

That makes sense, that should be the default behavior of the library, I opened that issue so we can implement that feature.

**Reply by williamschill** [reply]
I would like to see these two features in :NET MAUI as well. Here are some comments regarding the two:
- I would like to have the possibility to have the crosshair show/not show whenever I want. Maybe a bool property that you can set so that it is possible to control its visibility?
- Regarding the possibility to swipe the crosshair, this would be a huge improvement for the library and would make it very good on mobile devices where this is the preferred way of interacting with a chart. Here you should also be able to control the crosshairs visibility so it is possible for users to release their finger from the screen and still see where the crosshair is located.

Is there a link to the "swiping" issue so I can follow that one? I can only seem to find the one for "hiding crosshair"

**Reply by beto-rodriguez** [reply]
@williamschill in the next versions of the library we can use the `Axis.InvalidateCrosshair()` to show it (at a given position) and the `ClearCrosshair()` to hide it.

**Reply by kuznetsov-as** [reply]
@beto-rodriguez Can you please tell me when it will be possible to hide crosshairs when using winui3 I would be happy if the documentation had more information on crosshairs management, now it's completely unclear how to manage it. It would be cool if in the case of crosshairs everything would be as transparent as with tooltips
